src/create-model.jl

```diff
@@ -392,22 +448,41 @@ function create_model(
     ## Helper functions
     # Computes the duration of the `block` and multiply by the resolution of the
     # representative period `rp`.
-    function duration(timesteps_block, rp)
+    function duration(timesteps_block, rp, representative_periods)
         return length(timesteps_block) * representative_periods[rp].resolution
     end
+
+    Y = years = collect(keys(representative_periods)) # need to decide where to get the years
+
     # Maximum timestep
-    Tmax = maximum(last(rp.timesteps) for rp in representative_periods)
+    Tmax = maximum(last(rp.timesteps) for year in Y for rp in representative_periods[year])
+
     expression_workspace = Vector{JuMP.AffExpr}(undef, Tmax)
```

Comment:
how does it work in the creation of dataframes, when we don't add `year` to `expression_workspace`? because here the `expression_workspace` is only defined as the max timesteps within a rp

Reply:
The `expression_workspace` is a temporary pre-allocated array.
In the creation of dataframes, this is how it is used:

- Clean `workspace`
- Fill `workspace` looping over some partition
- Aggregate variables over some other partition and save then in the dataframe

So you don't need the `workspace` for anything else after that, so there is no need to store it for later, and so there is no need to separate per year, because they are completely ignored outside of the loop.
The only requirement is that the `workspace` is large enough to hold all the indices that the partition touch.